### DISCUSSION TITLE
ArgoCD apps in degraded state since upgrade to 2.12

### DISCUSSION THREAD

**Discussion by genseb13011** [argoproj/argo-cd#20716] [General]
Hi,

I'm facing an issue upgrading ArgoCD 2.12.

I'm currently using the argo-cd helm chart 7.3.11 (argocd version v2.11.7).

I tried to upgrade to helm chart 7.7.0 (v2.13.0) and noticed that apps related to the same helm chart switched to "degraded" state after upgrade.

I rollback to 7.3.11 and everything was well again.

I've notice that the error appears as soon as we switched to v2.12 version (7.4.0 helm chart).

Is there any changes in the 2.12 version which could explain this behaviour?

Thanks for your help.

Seb.

**Comment by genseb13011** [comment]
The issue is related to missing pods due to auto scaling.

I'll keep investigation and post how I fixed it.

Seb.

**Comment by andrii-korotkov-verkada** [comment]
Can you share more details, like screenshots, logs, events, please?

**Comment by genseb13011** [comment]
The issue is solved.

It seems that ArgoCD's healthcheck method has evolved between version and supports more resource types.

In my case it was a KafkaConnector kind that was with a bad status because of missing pods due to auto scaling process.

In previous version, ArgoCD didn't care about the KafkaConnector state, but from version 2.12, it does.

I've added below resource customizations (which is not perfect but do the job)

```
    resource.customizations.health.kafka.strimzi.io_KafkaConnector: |
      hs = {}
      if obj.status ~= nil then
        if obj.status.conditions ~= nil then
          for i, condition in ipairs(obj.status.conditions) do
            if condition.type == "NotReady" and condition.status == "False" then
              hs.status = "Degraded"
              hs.message = condition.message
              return hs
            end
            if condition.type == "Ready" and condition.status == "False" then
              hs.status = "Degraded"
              hs.message = condition.message
              return hs
            end
            if condition.type == "Ready" and condition.status == "True" then
              hs.status = "Healthy"
              hs.message = condition.message
              return hs
            end
            if condition.type == "NotReady" and condition.status == "True" then
              hs.status = "Healthy"
              hs.message = condition.message
              return hs
            end
          end
        end
      end
      hs.status = "Progressing"
      hs.message = condition.message
      return hs
```

Hope it could help.

Seb.
